packages/connect-sql/src/connect-sql-node.ts

```diff
@@ -0,0 +1,132 @@
+import { Connection, MetaStore } from '@fireproof/encrypted-blockstore'
+import { UploadDataFnParams, UploadMetaFnParams, DownloadDataFnParams, DownloadMetaFnParams, DBConnection, Store } from './types'
+
+import { DataRecord, DataSQLRecordBuilder, DataStoreFactory } from './data-type';
+import { MetaRecord, MetaRecordKey, MetaSQLRecordBuilder, MetaStoreFactory } from './meta-type';
+import { SQLFactory } from './sql';
+
+
+
+export class ConnectSQL extends Connection {
+  // readonly ws?: WebSocket
+  readonly decoder = new TextDecoder()
+
+  readonly dbConn: DBConnection
+
+  objectStore?: Store<DataRecord, string>
+  metaStore?: Store<MetaRecord, MetaRecordKey>
+
+  constructor(databaseURL: string) {
+    console.log('sql constructor', databaseURL);
+    super()
+    this.dbConn = SQLFactory(new URL(databaseURL))
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async dataUpload(bytes: Uint8Array, params: UploadDataFnParams) {
+    console.log('sql dataUpload', params);
+    await this.objectStore!.insert(
+      DataSQLRecordBuilder.fromUploadParams(bytes, params).build()
+    )
+    return Promise.resolve()
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async metaUpload(bytes: Uint8Array, params: UploadMetaFnParams): Promise<Uint8Array[] | null> {
+    console.log('sql metaUpload', params);
+    const ret = await this.metaStore!.insert(
+      MetaSQLRecordBuilder.fromUploadMetaFnParams(bytes, params).build()
+    )
+    return Promise.resolve(null)
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  dataDownload(params: DownloadDataFnParams): Promise<Uint8Array | null> {
+    console.log('sql dataDownload', params);
+    // const { type, name, car } = params
+    // const fetchFromUrl = new URL(`${type}/${name}/${car}.car`, this.downloadUrl)
+    // const response = await fetch(fetchFromUrl)
+    // if (!response.ok) return null // throw new Error('failed to download data ' + response.statusText)
+    // const bytes = new Uint8Array(await response.arrayBuffer())
+    // return bytes
+    return Promise.resolve(null)
+  }
+
+  async onConnect(): Promise<void> {
+    if (!this.loader || !this.taskManager) {
+      throw new Error('loader and taskManager must be set')
+    }
+    console.log('sql onConnect-1')
+    await this.dbConn.connect()
+    console.log('sql onConnect-2')
+    this.objectStore = await DataStoreFactory(this.dbConn)
+    console.log('sql onConnect-3')
+    this.metaStore = await MetaStoreFactory(this.dbConn)
+    console.log('sql onConnect-4')
+    return Promise.resolve()
+  }
+
+  /**
+   * metaDownload Function
+   *
+   * This function downloads metadata for a specific name and branch.
```

Comment:
I scrubbed this comment from the original. I think it might be wrong / confusing, so best to remove here also